styx-scheduler-service/src/main/java/com/spotify/styx/StateInitializingTrigger.java

```diff
@@ -50,13 +51,18 @@
   @Override
   public void event(Workflow workflow, Trigger trigger, Instant instant,
                     TriggerParameters parameters) {
-    if (workflow.configuration().dockerImage().isEmpty()) {
+    final var configuration = workflow.configuration();
+    if (configuration.flyteExecConf().isPresent()) {
+      LOG.info("{} is a Flyte Workflow, skipping as not supported at the moment", workflow.id());
+      return;
+    } else if (configuration.dockerImage().isEmpty()
```

Comment:
The flyteExecConf and dockerImage look unrelated at first.
Can we add a comment here describing the relation and why we need them in a else if block, instead of two separate early returns?
It should help us moving on when working with the code.

Reply:
Aha, yes good point. My tests didn't pass until I did the else if but I didn't document my finding